src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,27 @@
+import { useState } from 'react';
+
+export const useLocalStorage = <T>(
+  key: string,
+  initialValue: T
+): [T, (value: T) => void] => {
+  const [storedValue, setStoredValue] = useState<T>(() => {
+    try {
+      const item = window.localStorage.getItem(key);
+      return item ? JSON.parse(item) : initialValue;
```

Comment:
```suggestion
      return JSON.parse(item)
```
we don't need to check, we have `catch` which would handle error, if `item` isn't truthy